File: pythonCode/laplacianFilter.py
import cv2
import sys
import numpy as np

#read image
image = cv2.imread("../assets/crop1.png", cv2.IMREAD_GRAYSCALE)

#check if image exists
if image is None:
    print("can not find image")
    sys.exit()

#apply laplacian
laplacian = cv2.Laplacian(image, cv2.CV_32F, ksize=3, scale=1, delta=0)

#create lop kernel
logKernel = np.array(( [0.4038, 0.8021, 0.4038], [0.8021, -4.8233, 0.8021], [0.4038, 0.8021, 0.4038]), dtype="float")

#filter image using log kernel
logimg = cv2.filter2D(image, cv2.CV_32F, logKernel)

#normalize images
cv2.normalize(laplacian, laplacian, alpha=0, beta=1, norm_type=cv2.NORM_MINMAX, dtype=cv2.CV_32F)
cv2.normalize(logimg, logimg, alpha=0, beta=1, norm_type=cv2.NORM_MINMAX, dtype=cv2.CV_32F)

#create windows to display images
cv2.namedWindow("image", cv2.WINDOW_NORMAL)
cv2.namedWindow("laplacian", cv2.WINDOW_NORMAL)
cv2.namedWindow("log", cv2.WINDOW_NORMAL)

#display images
cv2.imshow("image", image)
cv2.imshow("laplacian", laplacian)
cv2.imshow("log", logimg)

#press esc to exit the program
cv2.waitKey(0)

#close all the opened windows
cv2.destroyAllWindows()

File: cPlusPlusCode/laplacianFilter.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace std;
using namespace cv;

int main(){

    //Read image
    Mat image = imread("../assets/crop1.png", IMREAD_GRAYSCALE);

    //check is image exits
    if(image.empty()){
        cout<<"can not find image"<<endl;
        return 0;
    }

    Mat laplacian, logimg;

    //Apply laplacian
    Laplacian(image, laplacian, CV_32F, 3, 1, 0);

    Mat LOGKernel = (Mat_<double>(3,3) <<  0.4038, 0.8021, 0.4038, 0.8021, -4.8233, 0.8021, 0.4038, 0.8021, 0.4038);

    //filter image using kernel
    filter2D(image, logimg, CV_32F, LOGKernel);

    //normalize images to display
    normalize(laplacian, laplacian, 0, 1, NORM_MINMAX);
    normalize(logimg, logimg, 0, 1, NORM_MINMAX);

    //create windows to display images
    namedWindow("image", WINDOW_NORMAL);
    namedWindow("laplacian", WINDOW_NORMAL);
    namedWindow("log", WINDOW_NORMAL);

    //show images
    imshow("image", image);
    imshow("laplacian", laplacian);
    imshow("log", logimg);
    
    //press esc to exit the program
    waitKey(0);

    //close all the opened windows
    destroyAllWindows();
    
    return 0;
}